src/main/java/duke/taskmanagement/TaskManager.java

```diff
@@ -0,0 +1,186 @@
+            int number = Integer.parseInt(taskNumber) - 1;
+            if (command.equals("mark")) {
+                taskList.get(number).setIsMarked();
+                System.out.println("Nice! I've marked this task as done:");
+            } else {
+                taskList.get(number).unsetIsMarked();
+                System.out.println("Nice! I've unmarked this task as done:");
+            }
+            taskRecorder.markOrUnmarkData(number);
+            System.out.println(number + 1 + "." + taskList.get(number).toString());
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            System.out.println("Oops! The task number given is not in range. Try again!");
+        } catch (NumberFormatException e) {
+            System.out.println("Oops! Please give an integer value. Try again!");
+        } catch (IOException e) {
+            System.out.println("Oops! Something went wrong writing to the file to mark or unmark a task!");
+        }
+    }
+
+    public void deleteTask(String userInput) {
+        try {
+            String[] words = userInput.split(" ");
+            String taskNumber = words[1];
+            int number = Integer.parseInt(taskNumber) - 1;
+            System.out.println("Noted. I've removed this task:\n" + taskList.get(number).toString());
+            taskList.remove(number);
+            taskRecorder.deleteData(number);
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            System.out.println("Oops! The task number given is not in range. Try again!");
+        } catch (NumberFormatException e) {
+            System.out.println("Oops! Please give an integer value. Try again!");
+        } catch (IOException e) {
+            System.out.println("Oops! Something went wrong with deleting a task from the file");
+        }
+    }
+
+    private String getDescription(String userInput) throws EmptyDescriptionException {
+        try {
+            int indexOfDescription = userInput.indexOf(" ");
+            int indexOfSlash = userInput.indexOf("/");
+            String description = "";
+            if (indexOfSlash != -1) {
+                description = userInput.substring(indexOfDescription, indexOfSlash);
+            } else {
+                description = userInput.substring(indexOfDescription);
+            }
+            if (description.equals(" ")) {
+                throw new EmptyDescriptionException();
+            }
+            return description;
+        } catch (StringIndexOutOfBoundsException | EmptyDescriptionException e) {
+            throw new EmptyDescriptionException();
+        }
+    }
+
+    private String getTimingDetails(String userInput) throws EmptyTimingDetailsException {
+        int indexOfSlash = userInput.indexOf("/");
+        if (indexOfSlash != -1) {
+            return userInput.substring(indexOfSlash + 4);
```

Comment:
[Magic numbers](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) like `4` here could be avoided by replacing them with named constants.